src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/HttpGoogleServicesDAO.scala

```diff
@@ -505,6 +526,50 @@ object HttpGoogleServicesDAO extends GoogleServicesDAO with FireCloudRequestBuil
     }
   }
 
+  override def checkGoogleGroupExists(groupEmail: String): Option[String] = {
+    val directoryService = getDirectoryManager(getDelegatedCredentialForAdminUser)
+    // check if group exists
+    val checkGroupRequest = directoryService.groups.get(groupEmail)
+    val existingGroup = Try(executeGoogleRequest(checkGroupRequest)) match {
+      case Failure(f) =>
+        logger.warn(s"Google Group $groupEmail does not exist")
+        "" // return nothing in case of failure
+      case Success(_) => {
+        groupEmail // return groupEmail
+      }
+    }
+    Option(existingGroup)
```

Comment:
Please forgive the drive by comment. I think this would be more idiomatic if it returned a boolean instead of an option-string. The caller already knows the group email, it doesn't necessarily need to know that again. Additionally, do you need to log the failure here in this underlying DAO method? Seems to me that information is more of a business logic and would be better handled in `UserService`.

Reply:
thank you for the very valuable drive by comment! great suggestions - made all these changes.